[27기 김태영] [Mission 1~4] 로그인 및 메인페이지 필수 구현사항 완료

Moro-yong opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 로그인 페이지 ID, PW validation 진행 중
- 메인페이지 좋아요 버튼 진행 중

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)

** 완료 기능 **
1. 필수기능

** 진행 중 **
1. 로그인 입력 데이터 유효성 검사
- id(pw)  value 값의 조건에 따라 id(pw) 인풋창 밑에 메시지 표시

2. 각 댓글 좋아요 버튼 
- 좋아요 버튼 클릭 이벤트 발생 시 색상 변경

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
- html과 js 파일을 불러오는 과정에서 js가 먼저 불러와질 경우 정상적인 작동이 안되는 것으로 defer 속성의 사용 이유에 대해 알게 되었습니다.
- css에 가상선택자를 미리 만들어 경우에 따라 가상선택자가 있는 것과 없는 것을 적용하는 방법이 아닌 disabled를 사용하는 방법을 배웠습니다.

<br />

## :: 기타 질문 및 특이 사항
- html 작성이 너무 지저분한거 같습니다 ㅠㅠ
- 자바스크립트 너무 어려워요;;

claire-euni commented on students/13th/jiankim/style/main.css:
```diff
@@ -0,0 +1,246 @@
+.feeds{
+    width: 612px;
+    margin-right: 12px;
+    flex-grow: 5;
```
width로 이미 지정값을 주었는데 flex-grow로 상대적인 값이 필요한가요?

claire-euni commented on students/13th/jiankim/style/main.css:
```diff
@@ -0,0 +1,246 @@
+    flex-grow: 5;
+}
+.feeds article {
+    display: flex;
+    flex-direction: column;
```
굳이 설정하지 않아도 세로 정렬이 되는데 필요한가요? :)

claire-euni commented on students/13th/jiankim/style/main.css:
```diff
@@ -0,0 +1,246 @@
+    border-radius: 50%;
+}
+.profileImage {
+    width: 30px;
+    height: 30px;
+    border-radius: 50%;
+    margin-right: 12px;
```
관련도 있는 css속성으로 묶어주신 것 베리굿👍

claire-euni commented on students/13th/jiankim/style/main.css:
```diff
@@ -0,0 +1,246 @@
+
+}
+.profileInfo img {
+
```
불필요한 엔터 삭제

claire-euni commented on students/13th/jiankim/main.html:
```diff
@@ -0,0 +1,305 @@
+    <script src="js/main.js"></script>
+    <script defer src="https://use.fontawesome.com/releases/v5.14.0/js/all.js" integrity="sha384-3Nqiqht3ZZEO8FKj7GR1upiI385J92VwWNLj+FqHxtLYxd9l+WYpeqSOrLh0T12c" crossorigin="anonymous"></script>
+  </body>
+</html>
```
전체적으로 잘 작성해주셨어요~다만 div보다는 semantic tag를 조금 더 사용해보시면 어떨까요? 그리고 불필요한 엔터들은 삭제해주세요 😄

claire-euni commented on students/13th/jiankim/js/main.js:
```diff
@@ -0,0 +1,17 @@
+const commentPostBtn = document.querySelector("#commentsPost");
+const commentList = document.getElementsByClassName('comments')[0];
+
+document.addEventListener('keyup',function() {
+    commentInput.value.length >= 1? commentPostBtn.style.opacity = "1" : commentPostBtn.style.opacity = "0.3";
```
이 부분도 login.js에 달린 리뷰처럼 간결하게 묶어볼수 있겠죠?

claire-euni commented on students/13th/jiankim/main.html:
```diff
@@ -0,0 +1,305 @@
+                <!-- input 여기에 추가 -->
+              </ul>
+              <form class="commentsContainer">
+                <input id="commentsInput" type="text" placeholder="댓글 달기...">
```
id는 고유값이라서 한번만 쓸 수 있습니다. 그런데 지금 input마다 동일 id를 갖고 있습니다. 수정해주세요~

claire-euni commented on students/13th/jiankim/js/main.js:
```diff
@@ -0,0 +1,17 @@
+const commentPostBtn = document.querySelector("#commentsPost");
+const commentList = document.getElementsByClassName('comments')[0];
+
+document.addEventListener('keyup',function() {
```
document에 event를 준 이유가 있을까요?

jian7061 commented on students/13th/jiankim/index.html:
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width">
+    <link rel="shortcut icon" href="img/favicon.ico" type="image/x-icon">
+    <title>Instagram</title>
+    <link href="style/reset.css" rel="stylesheet">
+    <link href="style.css" rel="stylesheet" type="text/css" />
+    <link href="style/common.css" rel="stylesheet">
+  </head>
+  <body>
+    <script src="script.js"></script>
+  </body>
+</html>
```
필요없습니다ㅎㅎ삭제했습니다!

jian7061 commented on students/13th/jiankim/login.html:
```diff
@@ -0,0 +1,95 @@
+    <link rel="shortcut icon" href="img/favicon.ico" type="image/x-icon">
+    <title>Instagram</title>
+    <link href="style/reset.css" rel="stylesheet">
+    <link href="style/common.css" rel="stylesheet" type="text/css" />
+     <link href="style/login.css" rel="stylesheet" type="text/css" />
+    <link href="style/common.css" rel="stylesheet">
```
수정했습니다~

jian7061 commented on students/13th/jiankim/login.html:
```diff
@@ -0,0 +1,95 @@
+      <div class="phoneImage">
+        <div class="slidingImages">
+          <img src="img/1.jpg">
+          <!-- <img src="img/2.jpg"> 
+          <img src="img/3.jpg">
+          <img src="img/4.jpg">
+          <img src="img/5.jpg">
+          <img src="img/1.jpg"> -->
```
녭

jian7061 commented on students/13th/jiankim/login.html:
```diff
@@ -0,0 +1,95 @@
+    <link href="style/common.css" rel="stylesheet" type="text/css" />
+     <link href="style/login.css" rel="stylesheet" type="text/css" />
+    <link href="style/common.css" rel="stylesheet">
+
```
오호 네!

jian7061 commented on students/13th/jiankim/login.html:
```diff
@@ -0,0 +1,95 @@
+        </div>
+      </div>
+      <div class="Container">
+        
```
수정했습니댜~~

Commit:
Add: 로그인 페이지 관련 파일 추가

Commit:
add : westargram main page

Commit:
Add : main.js
- 댓글 달기 기능 추가 (button click and Enter key press)

Modify : main.html, style.css
- 댓글 기능을 위해 태그 및 스타일 변경

Commit:
modify : 좋아요 버튼 진행 중
